kpi/views/v2/asset.py

```diff
@@ -618,6 +620,68 @@ def metadata(self, request):
         metadata = self.get_metadata(queryset)
         return Response(metadata)
 
+    @action(detail=False, methods=['GET'],
+            renderer_classes=[renderers.JSONRenderer],
+            url_path=r'migrate(?:/(?P<task_id>[\d\w\-]+))?')
+    def migrate(self, request, task_id: str = None, **kwargs):
+        """
+
+        Really basic endpoint what allows super users to migration projects
+        from KoBoCAT.
+
+        1. Call this endpoint with `?username=<username>`
+        2. Fetch url provided to check the state of the Celery task.
+           It can be:
+            - 'PENDING'
+            - 'FAILED'
+            - 'SUCCESS'
+
+        """
+        if not request.user.is_superuser:
+            raise Http404
```

Comment:
Not a big deal, but I think the rationale behind returning a 404 response when denying access is that revealing the existence of an endpoint would be a privacy concern in some cases. I'm not sure this is one of those cases, though.

Reply:
Ok, so 403 it will be :-)